include/deal.II/numerics/vector_tools_constraints.templates.h

```diff
@@ -533,30 +540,34 @@ namespace VectorTools
     using DoFToNormalsMap = std::multimap<
       internal::VectorDoFTuple<dim>,
       std::pair<Tensor<1, dim>,
-                typename DoFHandler<dim, spacedim>::active_cell_iterator>>;
+                typename DoFHandler<dim, spacedim>::cell_iterator>>;
     std::map<internal::VectorDoFTuple<dim>, Vector<double>>
       dof_vector_to_b_values;
 
     DoFToNormalsMap dof_to_normals_map;
 
     // now loop over all cells and all faces
-    typename DoFHandler<dim, spacedim>::active_cell_iterator
-      cell = dof_handler.begin_active(),
-      endc = dof_handler.end();
     std::set<types::boundary_id>::iterator b_id;
-    for (; cell != endc; ++cell)
-      if (!cell->is_artificial())
+    for (const auto &cell : dof_handler.cell_iterators_on_level(mesh_level))
+      if (cell->level_subdomain_id() != numbers::artificial_subdomain_id &&
+          cell->level_subdomain_id() != numbers::invalid_subdomain_id)
```

Comment:
Can this really work? You are now iterating only over the finest level, which does not comprise all active cells, as those might also be located on lower levels for adaptively refined meshes. I might be missing something, so I would be happy about some explanation.

Reply:
You are right. I will fix it, thanks.